### PR TITLE
Support building cncrypto lib with msvc

### PR DESCRIPTION
**Pull request by zone117x**
These changes allow building the hashing (cncrypto) lib on Windows with Visual Studio.

### CODE REVIEW

**Comment by zone117x** [comment]
If this gets merged I can also make a pull request with the visual studio project files for building the cncrypto dll https://github.com/zone117x/monero/commit/20a104adc73ebc5baa733939d6ce6f846555573e

File: src/crypto/chacha.c [hunk 1]
```diff
@@ -46,7 +48,7 @@ static void chacha(unsigned rounds, const void* data, size_t length, const uint8
   char* ctarget = 0;
   char tmp[64];
   int i;
-
+
```
Comment (moneromooo-monero) [inline src/crypto/chacha.c]: Please remove.

File: src/crypto/oaes_lib.c [hunk 2]
```diff
@@ -510,7 +510,7 @@ static uint32_t oaes_get_seed(void)
 	_test = (char *) calloc( sizeof( char ), timer.millitm );
 	_ret = gmTimer->tm_year + 1900 + gmTimer->tm_mon + 1 + gmTimer->tm_mday +
 			gmTimer->tm_hour + gmTimer->tm_min + gmTimer->tm_sec + timer.millitm +
-			(uintptr_t) ( _test + timer.millitm ) + getpid();
+			(uintptr_t) ( _test + timer.millitm ) + _getpid();
```
Comment (moneromooo-monero) [inline src/crypto/oaes_lib.c]: This should be only for _MSC_VER, right ?

Comment (zone117x) [reply src/crypto/oaes_lib.c]: You are correct - fixed

File: src/crypto/chacha.c (hunk 1 above)
Comment (zone117x) [reply src/crypto/chacha.c]: Fixed

File: src/common/int-util.h [hunk 3]
```diff
@@ -206,6 +209,12 @@ static inline void memcpy_swap64(void *dst, const void *src, size_t n) {
   }
 }
 
+#if _MSC_VER
```
Comment (moneromooo-monero) [inline src/common/int-util.h]: This is using #if, the previous one is using #ifndef, are you really trying to test for definedness in one case, and the value of it in the other ?

Comment (zone117x) [reply src/common/int-util.h]: Nice catch, fixed

**Pull request merged**